library/psa_crypto.c

```diff
@@ -2524,35 +2320,32 @@ static psa_status_t psa_mac_setup( psa_mac_operation_t *operation,
     psa_key_slot_t *slot;
     psa_key_usage_t usage =
         is_sign ? PSA_KEY_USAGE_SIGN_HASH : PSA_KEY_USAGE_VERIFY_HASH;
+    size_t mac_size = 0;
 
     /* A context must be freshly initialized before it can be set up. */
-    if( operation->alg != 0 )
-    {
+    if( operation->id != 0 )
         return( PSA_ERROR_BAD_STATE );
-    }
 
-    status = psa_mac_init( operation, alg );
-    if( status != PSA_SUCCESS )
-        return( status );
-    if( is_sign )
-        operation->is_sign = 1;
-
-    status = psa_get_and_lock_transparent_key_slot_with_policy(
+    status = psa_get_and_lock_key_slot_with_policy(
```

Comment:
As Ronald noted in #4464 this wasn't a bug: when a lock function fails, calling unlock is not required, but it's safe.

In this case I prefer to always jump to cleanup code. Having more consistent and systematic cleanup idioms helps with not forgetting important cleanup. But since this particular aspects is already inconsistent, I don't mind either way for this pull request.